src/test/SumsTests.scala

```diff
@@ -131,97 +201,24 @@ class SumsTests extends munit.FunSuite:
     assertEquals(res, Lefty())
   }
 
-  // TODO: not working - children of trait A were already queried before object Object was discovered.
-  // test("derive all subtypes in complex hierarchy") {
-  //   val res = Passthrough.derived[Complex].ctx.get.toOption.get
-  //   assertEquals(res.subtypes.size, 7)
-  //   List(
-  //     Complex.ObjectE,
-  //     Complex.Object,
-  //     Complex.Scoped.Object,
-  //     Complex.ClassH(1),
-  //     Complex.ObjectD,
-  //     Complex.ObjectF,
-  //     Complex.ObjectC
-  //   ).foreach { o =>
-  //     val chosen = res.choose(o)(identity)
-  //     assertEquals(chosen.value, o)
-  //     assertEquals(
-  //       chosen.typeInfo.short,
-  //       o.getClass.getSimpleName.replace("$", "")
-  //     )
-  //   }
-  // }
+   test("derive all subtypes in complex hierarchy") {
+     val res = Passthrough.derived[Complex].ctx.get.toOption.get
+
+     val pkg = "magnolia1.tests.SumsTests.Complex"
+     val expected = List(
+       s"$pkg.ClassH",
+       s"$pkg.Object",
+       s"$pkg.ObjectC",
+       s"$pkg.ObjectD",
+       s"$pkg.ObjectE",
+       s"$pkg.ObjectF",
+       s"$pkg.Scoped.Object",
+     )
+
+     assertEquals(res.subtypes.map(_.typeInfo.full).toList, expected)
```

Comment:
Yes, I changed test methodology a bit here, but the previous one was problematic as

```scala
     List(
       Complex.ObjectE,
       Complex.ObjectC,
       Complex.ObjectE,
       Complex.ObjectE,
       Complex.ObjectE,
       Complex.ObjectE,
       Complex.ObjectE
     ).foreach { o =>
       val chosen = res.choose(o)(identity)
       assertEquals(chosen.value, o)
       assertEquals(
         chosen.typeInfo.short,
         o.getClass.getSimpleName.replace("$", "")
       )
     }
```

passes tests just fine